src/storage-client/src/controller.rs

```diff
@@ -1205,11 +1223,7 @@ where
         // TODO: replace this with a new tokio task, rather than occupying
         // the existing read downgrader.
         let as_of = Antichain::from_elem(as_of);
-        self.state
-            .persist_read_handles
-            .snapshot(id, as_of)
-            .await
-            .unwrap()
+        self.state.persist_read_handles.snapshot(id, as_of).await
```

Comment:
genuine question (I don't know): is there any value in sending this to the worker vs just calling register/snap/expire here?

Reply:
Good question! I don't know why it was written that way but I agree that there is no point shipping the command to another async worker only to await its results. I'll move the logic in here since I'm working in this area